cpp/src/arrow/pretty_print.cc

```diff
@@ -377,6 +378,22 @@ class ArrayPrinter : public PrettyPrinter {
     return PrettyPrint(*array.indices(), ChildOptions(true), sink_);
   }
 
+  Status Visit(const RunEndEncodedArray& array) {
+    Newline();
+    Indent();
+    Write("-- run ends array (offset: ");
+    Write(std::to_string(array.offset()));
+    Write(", logical length: ");
+    Write(std::to_string(array.length()));
+    Write(")\n");
```

Comment:
Why write out the offset and length? We don't do this for dictionary indices, for example.

Reply:
Simplifying this.